java-tools/OsmAndServer/src/main/java/net/osmand/server/services/motd/MotdService.java

```diff
@@ -0,0 +1,219 @@
+package net.osmand.server.services.motd;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpHeaders;
+import org.springframework.stereotype.Service;
+import java.io.File;
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+
+@Service
+public class MotdService {
+    private static final Log LOGGER = LogFactory.getLog(MotdService.class);
+    private static final String MOTD_SETTINGS = "api/messages/motd_config.json";
+
+    @Value("${website.location}")
+    private String websiteLocation;
+
+    private final ObjectMapper mapper;
+
+    private MotdSettings settings;
+
+    @Autowired
+    public MotdService(ObjectMapper mapper) {
+        this.mapper = mapper;
+    }
+
+    public String getMessage(String version, String os, HttpHeaders headers) {
+        String message = null;
+        String hostAddress = getHostAddress(headers);
+        for (DiscountSetting setting : settings.getDiscountSettings()) {
+            if (handleCondition(setting.getDiscountCondition(), version, hostAddress)) {
```

Comment:
much cleanner will be to refactor to 
setting. isConditionTrue(version, hostAddress)